persistence/soft-index/src/main/java/org/infinispan/persistence/sifs/Index.java

```diff
@@ -342,17 +328,18 @@ private void shutdown() throws IOException {
          ByteBuffer buffer = ByteBuffer.allocate(4);
          buffer.putInt(0, freeBlocks.size());
          write(indexFile, buffer);
-         for (Map.Entry<Integer, List<IndexSpace>> entry : freeBlocks.entrySet()) {
+         for (Map.Entry<Short, List<IndexSpace>> entry : freeBlocks.entrySet()) {
             List<IndexSpace> list = entry.getValue();
             int requiredSize = 8 + list.size() * 10;
             buffer = buffer.capacity() < requiredSize ? ByteBuffer.allocate(requiredSize) : buffer;
             buffer.position(0);
             buffer.limit(requiredSize);
+            // TODO: change this to short
```

Comment:
Did you forget about this?

Reply:
No; I didn't want to change binary format in some micro - the plan is to add some version numbers both to index and to data files, and then I'll do the change to binary format as well.